### ISSUE TITLE
roll-up events for video detail page view and embed page view

### ISSUE THREAD

**Issue by pdpinch** [mitodl/odl-video-service#356]
In all cases I can think of, we want to combine video detail page events with embed page events, in order to look at a comprehensive count of plays and views. 

One way to roll these up would be to record embed page events using the same URL as the video page. 

https://s.nimbus.everhelper.me/attachment/1241215/dieufnzb01sn4av6mkxr/623857-7taGaRqAUiLUyp6I/screen.png

**Comment by mbertrand** [comment]
I think that using the `ga:pagePathLevel2` dimension in analytics should report combined events for both the video detail and video embed pages.  Here is a sample API body request for a particular video:
```
{
    'reportRequests': [
    {
      'viewId': VIEW_ID,
      'dateRanges': [{'startDate': '2017-12-21', 'endDate': '2017-12-23'}],
      'metrics': [
          {'expression': 'ga:uniqueEvents'},
          {'expression': 'ga:totalEvents'}
      ],        
      'dimensions': [{'name': 'ga:eventLabel'}, {'name': 'ga:pagePathLevel2'}],
      "dimensionFilterClauses": [{
          "filters": [{
              "dimensionName": "ga:pagePathLevel2",
              "operator": "EXACT",
              "expressions": ["/7dfff2ef1586418ea81f48eda833bf01/"]
          }]
      }]        
    }]
}
```

Another option is [manually overriding the page value on the embed page](https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications):
```javascript
ga('set', 'page', '/new-page.html');
ga('send', 'pageview');
```

**Comment by mbertrand** [comment]
Example comparing UI report results vs combined page API results:

![screen shot 2018-01-03 at 10 00 38 am](https://user-images.githubusercontent.com/187676/34525754-0571003c-f06e-11e7-9207-f45ab4b365c9.png)

```javascript
{
  "reports": [{
    "columnHeader": {
      "dimensions": ["ga:eventLabel",
        "ga:pagePathLevel2"
      ],
      "metricHeader": {
        "metricHeaderEntries": [{
            "name": "ga:uniqueEvents",
            "type": "INTEGER"
          },
          {
            "name": "ga:totalEvents",
            "type": "INTEGER"
          }
        ]
      }
    },
    "data": {
      "maximums": [{
        "values": ["2", "29"]
      }],
      "minimums": [{
        "values": ["2", "29"]
      }],
      "rowCount": 1,
      "rows": [{
        "dimensions": ["Play", "/7a3b729528624f548b5e4fee9fc44d79/"],
        "metrics": [{
          "values": ["2", "29"]
        }]
      }],
      "totals": [{
        "values": ["2", "29"]
      }]
    }
  }]
}
```

**Comment by mbertrand** [comment]
Look into changing the `embed` in the URL to a parameter instead (`?embed=true`)

**Comment by pdpinch** [comment]
After speaking with @Ferdi I don't think this is critical. I'm going to move it into the backlog.

**Comment by mbertrand** [comment]
I already implemented most of this yesterday (use embed as a parameter, and still have the old embed URL's work if anyone tried to use them) - but there's still tests to add or modify.  I can just save the branch for a future date though.

**Comment by pdpinch** [comment]
@mattbert I think you resolved this already?

**Comment by pdpinch** [comment]
@mbertrand I think you addressed this in #404

**Comment by mbertrand** [comment]
Yes, should be all set.
